### PR TITLE
refactor: DB independent quoting and truthy/falsy values

### PR DESCRIPTION
**Pull request by c-p-b**
Minor, straightforward changes to db independent syntax for table quoting, spacing, and truthy/falsy coercion

### CODE REVIEW

File: erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def create_sales_invoice_record(qty=1):
 			"due_date": today(),
 			"posting_date": today(),
 			"currency": "INR",
-			"taxes_and_charges": "",
+			"taxes_and_charges": '',
```
Comment (ankush) [inline erpnext/accounts/doctype/loyalty_program/test_loyalty_program.py]: This shouldn't be affecting it, it's python string :p

**Comment by ankush** [comment]
@cpdeethree can you once confirm the reason behind reformatting python strings to use single-quotes? 

We now autoformat code using black, so it will be turned back to double quotes by black unnless it's considered unsafe like `' "double quotes inside" '`. I don't think using single quote is a problem outside of SQL strings. 


reference: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#linting

**Comment by c-p-b** [comment]
As noted in our discussion, it looks like some of these changes happened when I originally went through and changed multiple double quoted values at once in the same file - even if it wasn't necessary. I will verify that there are no implications for postgres compatibility with the reformat you added (reverting back to the original)

**Comment by c-p-b** [comment]
Verified that PG has no problem with the changes you updated with the black commit 👍

File: erpnext/accounts/doctype/subscription/subscription.py [hunk 2]
```diff
@@ -148,8 +148,7 @@ def get_billing_cycle_and_interval(self):
 		billing_info = frappe.db.sql(
 			"select distinct `billing_interval`, `billing_interval_count` "
 			"from `tabSubscription Plan` "
-			"where name in %s",
-			(plan_names,),
+			"where name in {plan_names}".format(plan_names="('" + "','".join(plan_names) + "')"),
```
Comment (ankush) [inline erpnext/accounts/doctype/subscription/subscription.py]: Is this a problem with postgres `IN` queries? if so we should handle it in framework and not here

Comment (ankush) [reply erpnext/accounts/doctype/subscription/subscription.py]: created frappe issue: https://github.com/frappe/frappe/issues/17231

File: erpnext/assets/doctype/asset_maintenance/asset_maintenance.py [hunk 3]
```diff
@@ -54,8 +54,10 @@ def assign_tasks(asset_maintenance_name, assign_to_member, maintenance_task, nex
 	}
 	if not frappe.db.sql(
 		"""select owner from `tabToDo`
-		where reference_type=%(doctype)s and reference_name=%(name)s and status="Open"
-		and owner=%(assign_to)s""",
+		where reference_type=%(doctype)s and reference_name=%(name)s and status='Open'
+		and owner={assign_to}""".format(
+			assign_to="('" + team_member + "')"
+		),
```
Comment (ankush) [inline erpnext/assets/doctype/asset_maintenance/asset_maintenance.py]: Not sure why this change is required?

Comment (ankush) [reply erpnext/assets/doctype/asset_maintenance/asset_maintenance.py]: NVM this looks like it's buggy code 😮‍💨

File: erpnext/manufacturing/doctype/bom/bom.py [hunk 4]
```diff
@@ -1305,7 +1305,7 @@ def item_query(doctype, txt, searchfield, start, page_len, filters):
 		if not field in searchfields
 	]
 
-	query_filters = {"disabled": 0, "ifnull(end_of_life, '5050-50-50')": (">", today())}
+	query_filters = {"disabled": 0, "end_of_life": (">", today())}
```
Comment (ankush) [inline erpnext/manufacturing/doctype/bom/bom.py:1308]: This isn't correct. if end of life is not specified than it's considered to have infinite end of life.

File: erpnext/stock/report/incorrect_balance_qty_after_transaction/incorrect_balance_qty_after_transaction.py [hunk 5]
```diff
@@ -73,7 +73,7 @@ def get_stock_ledger_entries(report_filters):
 		"Stock Ledger Entry",
 		fields=fields,
 		filters=filters,
-		order_by="timestamp(posting_date, posting_time) asc, creation asc",
+		order_by="posting_date asc, posting_time asc, creation asc",
```
Comment (ankush) [inline erpnext/stock/report/incorrect_balance_qty_after_transaction/incorrect_balance_qty_after_transaction.py:76]: @cpdeethree I think this is fine 

however in the past, people have suggested relying on the timestamp function instead of sorting columns individually, do you know of any real differences between these two?

File: erpnext/stock/doctype/warehouse/warehouse.py [hunk 6]
```diff
@@ -161,8 +161,8 @@ def get_children(doctype, parent=None, company=None, is_root=False):
 
 	fields = ["name as value", "is_group as expandable"]
 	filters = [
-		["docstatus", "<", "2"],
-		['ifnull(`parent_warehouse`, "")', "=", parent],
+		["ifnull(`docstatus`, 0)", "<", 2],
```
Comment (ankush) [inline erpnext/stock/doctype/warehouse/warehouse.py]: IMO this condition doesn't make sense. The warehouse has no canceled status 😄

**Review by ankush — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/subscription/subscription.py (hunk 2 above)
Comment (ankush) [reply erpnext/accounts/doctype/subscription/subscription.py]: Fixed by rewriting query

File: erpnext/stock/doctype/warehouse/warehouse.py (hunk 6 above)
Comment (ankush) [reply erpnext/stock/doctype/warehouse/warehouse.py]: removed this

**Review by ankush — DISMISSED** [review]
